src/saveload.c

```diff
@@ -292,28 +292,25 @@ void SaveSubObj( Obj subobj )
     SaveUInt((UInt) subobj);
   else if (IS_FFE(subobj))
     SaveUInt((UInt) subobj);
-  else if ((((UInt)subobj & 3) != 0) || 
-           subobj < (Bag)MptrBags || 
-           subobj > (Bag)MptrEndBags ||
-           (Bag *)PTR_BAG(subobj) < MptrEndBags)
+  else if (IS_BAG_ID(subobj))
```

Comment:
Is this the same? It seems to have dropped the `PTR_BAG(subobj) < MptrEndBags` part? Which I think I've seen occasionally trigger when memory corruption bugs occur.

Reply:
I've now adjusted the test to do this once again, too